src/components/LayoutWrapper/LayoutWrapper.tsx

```diff
@@ -16,6 +20,19 @@ import {
 const LayoutWrapper = ({ children }) => {
   const navigate = useNavigate()
   const location = useLocation()
+  const dispatch = useDispatch()
+  const user = useAppSelector((state) => state.user)
+  const localStorageUser = JSON.parse(localStorage.getItem('user'))
+
+  const { setUser } = bindActionCreators(actionCreators, dispatch)
+
+  useEffect(() => {
+    if (localStorageUser === null) {
+      navigate('/auth')
+    } else {
+      setUser(localStorageUser.user)
```

Comment:
nonononononononono, it will create so many problems that i'll better stay silent

Reply:
Any thoughts how it can be fixed?